plugins/modules/web_infrastructure/jira.py

```diff
@@ -662,35 +410,263 @@ def main():
         supports_check_mode=False
     )
 
-    op = module.params['operation']
-
-    # Handle rest of parameters
-    uri = module.params['uri']
-    user = module.params['username']
-    passwd = module.params['password']
-    if module.params['assignee']:
-        module.params['fields']['assignee'] = {'name': module.params['assignee']}
-    if module.params['account_id']:
-        module.params['fields']['assignee'] = {'accountId': module.params['account_id']}
-
-    if not uri.endswith('/'):
-        uri = uri + '/'
-    restbase = uri + 'rest/api/2'
+    state_param = 'operation'
+
+    def __init_module__(self):
+        if self.vars.fields is None:
+            self.vars.fields = {}
+        if self.vars.assignee:
+            self.vars.fields['assignee'] = {'name': self.vars.assignee}
+        if self.vars.account_id:
+            self.vars.fields['assignee'] = {'accountId': self.vars.account_id}
+        self.vars.uri = self.vars.uri.strip('/')
+        self.vars.set('restbase', self.vars.uri + '/rest/api/2')
+
+    @cause_changes(on_success=True)
+    def operation_create(self):
+        createfields = {
+            'project': {'key': self.vars.project},
+            'summary': self.vars.summary,
+            'issuetype': {'name': self.vars.issuetype}}
+
+        if self.vars.description:
+            createfields['description'] = self.vars.description
+
+        # Merge in any additional or overridden fields
+        if self.vars.fields:
+            createfields.update(self.vars.fields)
+
+        data = {'fields': createfields}
+        url = self.vars.restbase + '/issue/'
+        self.vars.meta = self.post(url, self.vars.username, self.vars.password, self.vars.timeout, data)
+
+    @cause_changes(on_success=True)
+    def operation_comment(self):
+        data = {
+            'body': self.vars.comment
+        }
+        url = self.vars.restbase + '/issue/' + self.vars.issue + '/comment'
+        self.vars.meta = self.post(url, self.vars.username, self.vars.password, self.vars.timeout, data)
+
+    @cause_changes(on_success=True)
+    def operation_edit(self):
+        data = {
+            'fields': self.vars.fields
+        }
+        url = self.vars.restbase + '/issue/' + self.vars.issue
+        self.vars.meta = self.put(url, self.vars.username, self.vars.password, self.vars.timeout, data)
+
+    @cause_changes(on_success=True)
+    def operation_update(self):
+        data = {
+            "update": self.vars.fields,
+        }
+        url = self.vars.restbase + '/issue/' + self.vars.issue
+        self.vars.meta = self.put(url, self.vars.username, self.vars.password, self.vars.timeout, data)
+
+    @cause_changes(on_success=True)
+    def operation_fetch(self):
+        url = self.vars.restbase + '/issue/' + self.vars.issue
+        self.vars.meta = self.get(url, self.vars.username, self.vars.password, self.vars.timeout)
+
+    @cause_changes(on_success=True)
```

Comment:
Is this correct? This function did not returned `True` for changed in the old version.